WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileLayout.kt

```diff
@@ -0,0 +1,177 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.wear.tiles.messaging
+
+import android.content.Context
+import android.graphics.BitmapFactory
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.wear.tiles.DeviceParametersBuilders
+import androidx.wear.tiles.ModifiersBuilders
+import androidx.wear.tiles.material.Button
+import androidx.wear.tiles.material.ButtonColors
+import androidx.wear.tiles.material.ChipColors
+import androidx.wear.tiles.material.CompactChip
+import androidx.wear.tiles.material.layouts.MultiButtonLayout
+import androidx.wear.tiles.material.layouts.PrimaryLayout
+import com.example.wear.tiles.R
+import com.example.wear.tiles.tools.IconSizePreview
+import com.example.wear.tiles.tools.WearSmallRoundDevicePreview
+import com.example.wear.tiles.tools.emptyClickable
+import com.google.android.horologist.compose.tools.LayoutElementPreview
+import com.google.android.horologist.compose.tools.LayoutRootPreview
+import com.google.android.horologist.compose.tools.buildDeviceParameters
+import com.google.android.horologist.tiles.images.drawableResToImageResource
+
+/**
+ * Layout definition for the Messaging Tile.
+ *
+ * By separating the layout completely, we can pass fake data for the [MessageTilePreview] so it can
+ * be rendered in Android Studio (use the "Split" or "Design" editor modes).
+ */
+internal fun messagingTileLayout(
+    state: MessagingTileState,
+    context: Context,
+    deviceParameters: DeviceParametersBuilders.DeviceParameters
+) = PrimaryLayout.Builder(deviceParameters)
+    .setContent(
+        MultiButtonLayout.Builder()
+            .apply {
+                // In a PrimaryLayout with a compact chip at the bottom, we can fit 5 buttons.
+                // We're only taking the first 4 contacts so that we can fit a Search button too.
+                state.contacts.take(4).forEach { contact ->
+                    addButtonContent(contactLayout(context, contact, emptyClickable))
+                }
+            }
+            .addButtonContent(searchLayout(context, emptyClickable))
+            .build()
+    ).setPrimaryChipContent(
+        CompactChip.Builder(
+            context,
+            context.getString(R.string.tile_messaging_create_new),
+            emptyClickable,
+            deviceParameters
+        )
+            .setChipColors(ChipColors.primaryChipColors(MessagingTileTheme.colors))
```

Comment:
this ok?

Reply:
Yeah, I think it's the right approach for samples, unless we want to advocate for dynamically changing themes.  Maybe for media, we should show that when artwork changes?

Reply:
Tangential: related to the repetition of context, context.getString(), deviceParameters, MessagingTileTheme.colors. This was something that I wanted to improve by having a Renderer type that could simplify these repetitions. https://github.com/google/horologist/blob/ab8ae486d43b55e46423979d9ad2aacf382d323c/tiles/src/main/java/com/google/android/horologist/tiles/render/SingleTileLayoutRenderer.kt

But not for here, maybe something we can review these and get a nice streamlined API.

Reply:
>unless we want to advocate for dynamically changing themes. Maybe for media, we should show that when artwork changes?

yeah let's see how it looks in practice